lib/request.js

```diff
@@ -0,0 +1,102 @@
+'use strict'
+const http = require('http')
+const https = require('https')
+const http2 = require('http2')
+const eos = require('end-of-stream')
+const pump = require('pump')
+const { stripHttp1ConnectionHeaders } = require('./utils')
+
+class RequestAgent {
```

Comment:
I'm very -1 on having this as a class. Can you restore the approach with just functions?

Something like:


```js
function buildRequest (opts) {
  return { request, close }

  function request () {}
  function close () {}
}
```